spinn_front_end_common/interface/interface_functions/pre_allocate_for_bit_field_router_compressor.py

```diff
@@ -37,24 +36,17 @@ def __call__(self, machine, pre_allocated_resources):
         :rtype: ~pacman.model.resources.PreAllocatedResourceContainer
         """
 
-        progress_bar = ProgressBar(
-            machine.n_chips,
-            "Preallocating resources for bit field compressor")
-
         # for every Ethernet connected chip, get the resources needed by the
         # live packet gatherers
-        sdrams = list()
         sdram_to_pre_alloc_for_bit_fields = get_config_int(
             "Mapping",
             "router_table_compression_with_bit_field_pre_alloced_sdram")
-        for chip in progress_bar.over(machine.chips):
-            sdram = ConstantSDRAM(
-                (SIZE_OF_SDRAM_ADDRESS_IN_BYTES * chip.n_user_processors) +
-                sdram_to_pre_alloc_for_bit_fields)
-            sdrams.append(SpecificChipSDRAMResource(chip, sdram))
+        sdram = ConstantSDRAM(
+            (SIZE_OF_SDRAM_ADDRESS_IN_BYTES * Machine.max_cores_per_chip()) +
+            sdram_to_pre_alloc_for_bit_fields)
 
         # note what has been preallocated
-        allocated = PreAllocatedResourceContainer(
-            specific_sdram_usage=sdrams)
-        allocated.extend(pre_allocated_resources)
-        return allocated
+        pre_allocated_resources.add_sdram_all(sdram)
+
+        progress_bar.end()
```

Comment:
This line is no longer wanted; it makes the code fail.

Reply:
agreed Gone!